### PR TITLE
Add support for multi-lines prompt.

### PR DESCRIPTION
**Pull request by ralt**
This is useful for e.g. pinentry that wants to ask for a passphrase
after explaining which key needs to be unlocked.

### CODE REVIEW

File: input.lisp [hunk 1]
```diff
@@ -288,14 +302,18 @@ match with an element of the completions."
...
                                       (string char)
                                       :translate #'translate-id
                                       :size 16))
             else
               do (draw-image-glyphs win gcontext (screen-font screen)
                                     x
-                                    (font-ascent (screen-font screen))
+                                    (* (+ *message-window-padding*
+                                          (font-ascent (screen-font screen)))
+                                       prompt-lines-length)
```
Comment (Harleqin) [inline input.lisp]: That is far enough repetition to warrant a separate function `text-height`.

Comment (dbjergaard) [reply input.lisp]: abstract this into a function and update the calls to it and I'll merge this PR.

Comment (ralt) [reply input.lisp]: I guess I need to rebase on top of #404 too, which should solve the repetition issue.

**Comment by ralt** [comment]
@Harleqin @dbjergaard I mostly rewrote the PR after wanting to rebase on top of #404 (conflicts, woo!), which led me to a better understanding of the code, and the new result is much better than the original version, imho.

Please review :)

**Comment by ralt** [comment]
Some quick testing can be done with the following in the REPL:

```
(read-one-line (current-screen) "foo: ")
(read-one-line (current-screen) "foo
bar: "))
(read-one-line (current-screen) "foo
bar

baz: "))
(let ((*message-window-y-padding* 50))
  (read-one-line (current-screen) "foo
bar: ")))
```

**Comment by PuercoPop** [comment]
@ralt Awesome!

**Pull request merged**